### ISSUE TITLE
lpoptions(1) `-a' (all printers) option

### ISSUE THREAD

**Issue by michaelrsweet** [apple/cups#2082]
Version: 1.5-feature
CUPS.org User: murdie

I would like to see a `-a' (`all printers') option for lpoptions(1). This would cause the remaining options on the command line to become the default for all printers (usually for oneself, but for all users if run as superuser - as always for lpoptions(1)). E.g.

$ lpoptions -a -o page-left=20 -o page-top=20

might enter:

All page-left=20 page-top=20

in the lpoptions file.

The use of the -d/-p options with -a in the same command is an error.

Default options set with `-a' might accumulate and/or replace each other as they do now, so:

$ lpoptions -a -o copies=2
$ lpoptions -a -o copies=3

would result in:

All copies=3

and a further:

$ lpoptions -a -o number-up=2

would result in:

All copies=3 number-up=2

in the lptotions file. A subsequent:

$ lpoptions -a -r number-up

would reduce the lpoptions file back to:

All copies=3

Presumably one needs an `lpoptions -r all' option also!

Should individually-set default options be overridden by `-a'? E.g. if the lpoptions file contained:

Default p1 copies=2
Dest p2 copies=3

then should the command:

$ lpoptions -a -o copies=4

change it to:

Default p1
All copies=4

Or should an error message be issued (because the `-a -o copies=4' would override the existing settings)?

What if the lpoptions file does contain the above, and we command:

$ lpoptions -p p3 -o copies=1

Should the `copies=1' for printer`p3' override the `copies=4' for all printers, or should the command above should be an error (because`-a copies=4' already exists)?

In other words, is the `-a' default a sort of`default-default' or something that cannot override an existing setting? Which is the most useful, least confusing, behaviour?

John A. Murdie
